src/check_jsonschema/formats/implementations/rfc5321.py

```diff
@@ -0,0 +1,41 @@
+import re
+
+RFC5321_REGEX = re.compile(
+    r"""
+    ^
+    (
+    [!#-'*+/-9=?A-Z^-~-]+(\.[!#-'*+/-9=?A-Z^-~-]+)*
+    |
+    "([]!#-[^-~ \t]|(\\[\t -~]))+"
+    )
+    @
+    (
+    [!#-'*+/-9=?A-Z^-~-]+(\.[!#-'*+/-9=?A-Z^-~-]+)*
+    |
+    \[[\t -Z^-~]*]
+    )
+    $
+    """,
```

Comment:
I'm finding this regex a bit difficult to read. In particular, I'm seeing some unconventional range expressions in the character classes, like `/-9`, ` -Z`, and `^-~`.
These are valid, but they aren't the way character classes are typically written. Perhaps other people have an expert and intuitive knowledge of what `chr(ord(" ")+1)` will be, but I definitely don't.

Can these be rewritten such that the suite of characters matched is more obvious to a reader? For example, rather than `[/-9]`, I would much rather see `[/0-9]`. The fact that the lowercase letters are captured with `^-~` caught me particularly off-guard.

Reply:
That's fair. I'm not the original author of these regexes either (links in PR description). I'll see what I can do about cleaning up those character classes though.

Reply:
I had some time this week to revisit this, and reworked the simpler RFC5321 validation. I also added in some length checks as well and tried validating it against the examples in this wikipedia page: https://en.wikipedia.org/wiki/Email_address#Examples

It validated/invalidated as expected except for a couple cases:
- `I❤️CHOCOLATE@example.com` was incorrectly found invalid, likely because of UTF-8
- `i.like.underscores@but_they_are_not_allowed_in_this_part` was incorrectly found invalid, we allow underscores in the domain part of the regex.

Let me know if the regex is easier to understand and if you think we should need to handle non-ascii strings like that utf-8 one.